build-system/tasks/e2e/mocha-custom-json-reporter.js

```diff
@@ -35,13 +35,13 @@ const {inherits} = require('mocha').utils;
  */
 async function writeOutput(output, filename) {
   try {
-    await fs.mkdir('result-reports');
+    await fs.ensureDir('result-reports');
```

Comment:
Since we're using `fs-extra`, `outputJSON` should be used here. Handles both JSON serialization and creates missing directories.

https://github.com/jprichardson/node-fs-extra/blob/master/docs/outputJson.md

Reply:
Good idea. Done.